src/BenchmarkDotNet.Core/Running/BenchmarkRunnerCore.cs

```diff
@@ -366,7 +366,7 @@ private static BenchmarkReport RunCore(Benchmark benchmark, ILogger logger, Read
             {
                 logger.WriteLineInfo("// Run, Diagnostic [SeparateLogic]");
 
-                separateLogicCompositeDiagnoser.Handle(HostSignal.AfterAll, new DiagnoserActionParameters(null, benchmark, config));
+                separateLogicCompositeDiagnoser.Handle(HostSignal.SeparateLogic, new DiagnoserActionParameters(null, benchmark, config));
```

Comment:
@morgan-kn very good catch! you have fixed my bug 💯

Reply:
@adamsitnik thanks =)